clang/lib/Sema/SemaOpenACC.cpp

```diff
@@ -63,17 +67,17 @@ void Sema::ActOnOpenACCConstruct(OpenACCDirectiveKind K,
     // here as these constructs do not take any arguments.
     break;
   default:
-    Diag(StartLoc, diag::warn_acc_construct_unimplemented) << K;
+    Sema.Diag(StartLoc, diag::warn_acc_construct_unimplemented) << K;
```

Comment:
I've seen this enough... I wonder if the 'sub' types should have a 'Diag' reference as well?  WDYT?

Reply:
We can do that, but that's far from the only piece of "common infrastructure" in `Sema`. Do we also want `LangOpts`, `ASTContext`, `SourceMgr`, etc. in `Sema` components? If so, where do we stop?

Obviously, this is going to cost us some memory to store those references, and some boilerplate in constructors of components to initialize them.

Reply:
Hmm... thats a good point.  Perhaps `Diag` is just over represented here.